lib/dispatch.go

```diff
@@ -29,6 +30,14 @@ type Cursor interface{}
 // with the context.Context replaced by a scope.
 type MethodSet interface {
 	Name() string
+	Attributes() map[string]AttributeSet
+}
+
+// AttributeSet is extra information about each method, such as: http endpoint,
+// http verb, (TODO) permissions, and (TODO) other metadata
+type AttributeSet struct {
```

Comment:
are attributes _required_ for all methods defined on `MethodSet` implementers? This commend should clarify.

Reply:
Yes, the reasoning being that it's easier to require all methods to have attributes, rather than allow misspellings to go unnoticed without error.